adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -2531,6 +2546,23 @@ abstract class AvroGenomicRDD[T <% IndexedRecord: Manifest, U <: Product, V <: A
     saveSequences(filePath)
   }
 
+  protected def referenceRegionsToDatasetQueryString(regions: Iterable[ReferenceRegion], partitionSize: Int = 1000000, partitionedLookBackNum: Int = 1): String = {
```

Comment:
I see that a lot of downstream classes that implement `GenomicDataset` wind up overriding `filterDatasetByOverlappingRegions` with `transformDataset(ds => ds.filter(referenceRegionsToDatasetQueryString(...))`. I get why they have to do this, and I think fixing it is out of scope for this PR. However, can you open a ticket for us to move that code into a trait in a future release?

Reply:
In update the `filterDatasetByOverlappingRegion` is implemented generically in trait `DatasetBoundGenomicDataset` , thus no more overrideing